oak-benchmarks-elastic/src/main/java/org/apache/jackrabbit/oak/benchmark/ElasticBenchmarkRunner.java

```diff
@@ -25,7 +25,7 @@ public class ElasticBenchmarkRunner extends BenchmarkRunner {
 
     private static ElasticConnection connection;
 
-    public static void main(String[] args) throws Exception {
+    public static int main(String[] args) throws Exception {
```

Comment:
You can run Java programs with "java ... <className>" but only if they have public static *void* main method. So this change would have the (unintended?) consequence that this is no longer possible.

I think public static void main, and then System.exit(code) wouldn't cause this incompatibility. Are you sue System.exit doesn't work?

Reply:
Actually... With System.exit(code), behaviour of existing code would change it it calls the main method from some other method... so instead of System.exit(code > 0), maybe throwing an exception would be the right way?

Reply:
```ElasticBenchmarkRunner``` is called from the BenchmarkElasticCommand, and it does not integrate into ```oak-run```, but nevertheless implements ```Command```.

I agree, you should still have the chance to start the  ```ElasticBenchmarkRunner``` as a standalone command, but in that case I would move the logic out of the ```main``` method into a ```runBenchmark``` method, which gets the same params as ```main``` but has a return value. 

Will post an update to this PR.

Reply:
Whenever a command executes ```System.exit(statuscode);``` directly, a stacktrace will be written during termination. For the shutdownHook in ```org.apache.jackrabbit.oak.run.Main```  it's not possible to find out if the JVM is terminated by the OS or by triggered by a ```System.exit()``` call.
Therefor all commands need to return to this main method.

Reply:
Regarding the "throw an exception instead of returning a int as exitcode": This could be an alternative, but I don't see any improvement over my approach. Because if you just throw an exception and let the JVM handle it, the JVM will implicitly set the exit code to "1". 
This would allow a binary state: Success or fail. But looking at the variety of calls to ```System.exit()``` in the various commands, I see more exit codes than just "0" and "1". I believe we even have "-1". 
Converting these cases all into throwing exceptions, we would also get changes in the behavior. Not to mention that java terminating a program with an exception is likely to change the output of the tool.